tower-http/src/classify.rs

```diff
@@ -351,7 +402,8 @@ mod usable_for_retries {
 
     impl<ReqB, ResB, E, C> Policy<Request<ReqB>, Response<ResB>, E> for RetryBasedOnClassification<C>
     where
-        C: ClassifyResponse<E> + Clone,
+        C: ClassifyResponse + Clone,
+        E: fmt::Display + 'static,
```

Comment:
is Error a better bound here? since it includes display?

Reply:
I wish we could require `std::error::Error` but since `Box<dyn std::error::Error>` doesn't implement `std::error::Error` it wouldn't work 😞

Reply:
lovely...hahah..sounds good if it compiles etc I am happy with it